vendors/st/boards/stm32l475_discovery/ports/common_io/iot_spi.c

```diff
@@ -279,7 +301,26 @@ int32_t iot_spi_ioctl( IotSPIHandle_t const pxSPIPeripheral,
                             break;
                     }
 
-                    LL_SPI_SetBaudRatePrescaler( pxSpi->Instance, ( ( IotSPIMasterConfig_t * ) pvBuffer )->ulFreq );
+                    uint32_t sysClkTmp = SystemCoreClock;
+                    uint32_t preScaler = 0;
+                    uint8_t divisor = 0;
+
+                    while( sysClkTmp > ( ( IotSPIMasterConfig_t * ) pvBuffer )->ulFreq )
```

Comment:
This would be clearer if written as:

```c
while( ( sysClkTmp > ( ( IotSPIMasterConfig_t * ) pvBuffer )->ulFreq ) && ( divisor < 7 ) )
{
    divisor++;
    sysClkTmp = ( sysClkTmp >> 1 );
}	
```

Reply:
That sounds better. Will do.